marklogic-data-hub-central/ui/src/components/related-entities-facet/related-entities-facet.tsx

```diff
@@ -17,24 +18,48 @@ interface Props {
   onSettingCheckedList: (checkAll: any) => void;
   setCurrentRelatedEntities: (relatedEntities: Map<string, any>) => void;
   setEntitySpecificPanel: (entity: any) => void;
+  setActiveRelatedEntities: (boolean) => void;
 }
 
 const RelatedEntitiesFacet: React.FC<Props> = (props) => {
 
   const {
+    searchOptions,
     setRelatedEntityTypeIds
   } = useContext(SearchContext);
   const {currentRelatedEntities, onSettingCheckedList, setCurrentRelatedEntities, setEntitySpecificPanel} = props;
   const [entitiesList, setEntitiesList] = useState<any[]>([]);
   const [showMore, setShowMore] = useState<boolean>(false);
   const [options, setOptions] = useState<any[]>([]);
   const [checkedList, setCheckedList] = useState<any[]>([]);
+  const [relatedEntitiesDisabled, setRelatedEntitiesDisabled] = useState<any[]>([]);
 
   useEffect(() => {
     const entityNames = Array.from(currentRelatedEntities.keys());
     setEntitiesList(entitiesSorting(entityNames));
   }, [currentRelatedEntities, checkedList]);
 
+  useEffect(() => {
+    Array.from(currentRelatedEntities.keys()).forEach((entity) => {
+      if (selectedInBase(entity) && !relatedEntitiesDisabled.includes(entity)) {
+        setRelatedEntitiesDisabled(prevState => [...prevState, entity]);
```

Comment:
Since we have a useEffect below that gets triggered for each item we are adding/updating to relatedEntitiesDisabled, wouldn't it be better to get a temp list and add all new disabled only one time after the array loop ends?

Reply:
@aniamy Good point, I agree. Pushed the recommended fix.